enumerables.rb

```diff
@@ -0,0 +1,117 @@
+module Enumerable
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i])
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i], i)
+    end
+    self
+  end
+
+  def my_select()
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |item|
+      new_array << item if yield(item) == true
+    end
+    new_array.empty? ? self : new_array
```

Comment:
[Mandatory]
Can you add a scenario where it returns self(In slack please, really curious to know it)? Cause I don't know when it returns self. As far I know it returns a new array based on the given block. If the block is not given return an enumerator.